p2p/protocol/circuitv2/relay/relay.go

```diff
@@ -556,6 +678,8 @@ func (r *Relay) background() {
 		case <-ticker.C:
 			r.gc()
 		case <-r.ctx.Done():
+			r.gc()
+			r.wg.Done()
```

Comment:
Was this a bug in our implementation? If so, would you mind opening a separate PR?

Reply:
it's a race condition. yes. Will do. https://github.com/libp2p/go-libp2p/issues/2162

Reply:
This is the fix for the bug. I added the gc call there since it simplified counting closed connections. 
https://github.com/libp2p/go-libp2p/pull/2164/files